gateway/src/main/kotlin/com/gitlab/kordlib/gateway/Intent.kt

```diff
@@ -149,83 +152,97 @@ enum class Intent(val code: Int) {
  * A set of [intents][Intent] to be used while [identifying][Identify] a [Gateway] connection to communicate the events the client wishes to receive.
  */
 @Serializable(with = IntentsSerializer::class)
-data class Intents internal constructor(val code: Int) {
+data class Intents internal constructor(var code: Int = 0) {
 
     val intents = Intent.values().filter { code and it.code != 0 }.toSet()
 
+
     operator fun contains(intent: Intent) = intent in intents
 
     /**
      * Returns an [Intents] that added the [intent] to this [code].
      */
-    operator fun plus(intent: Intent): Intents = when {
-        code and intent.code == intent.code -> this
-        else -> Intents(this.code or intent.code)
-    }
+    operator fun plus(intent: Intent): Intents = Intents(this.code or intent.code)
+
+
+    /**
+     * Returns an [Intents] that added [code]s  in [Intents] to this [code].
+     */
+    operator fun plus(intent: Intents): Intents = Intents(this.code or intent.code)
+
+    /**
+     * Returns an [Intents] that removed the [intent] from this [code].
+     */
+    operator fun minus(intent: Intent): Intents = Intents(code xor intent.code)
 
     /**
      * Returns an [Intents] that removed the [intent] from this [code].
      */
-    operator fun minus(intent: Intent): Intents = when {
-        code and intent.code == intent.code -> Intents(code xor intent.code)
-        else -> this
+
+    operator fun minus(intent: Intents): Intents = Intents(code xor intent.code)
+
+
+
+
+    /**
+     * adds this intent to the [code] of an [Intents]
+     */
+
+    operator fun Intent.unaryPlus(): Intents {
+        return plus(this)
     }
 
     /**
-     * copy this [Intents] and apply the [block] to it.
+     * removes this intent from the [code] of an [Intents]
      */
-    @OptIn(ExperimentalContracts::class)
-    inline fun copy(block: IntentsBuilder.() -> Unit): Intents {
-        contract {
-            callsInPlace(block, InvocationKind.EXACTLY_ONCE)
-        }
-        val builder = IntentsBuilder(code)
-        builder.apply(block)
-        return builder.flags()
+
+
+    operator fun Intent.unaryMinus(): Intents {
+        return minus(this)
+    }
+
+
+    /**
+     * adds this intent to the [code] of an [Intents]
+     */
+    operator fun Intents.unaryPlus(): Intents {
+        return plus(this)
     }
 
+
+    /**
+     * removes this intent from the [code] of an [Intents]
+     */
+
+    operator fun Intents.unaryMinus(): Intents {
+        return minus(this)
+    }
+
+
     companion object {
 
         @PrivilegedIntent
         val all: Intents
-            get() = invoke {
-                Intent.values().forEach { +it }
-            }
+            get() = Intents(*Intent.values())
```

Comment:
The spread operator is all kinds horrible in terms of performance: https://sites.google.com/a/athaydes.com/renato-athaydes/posts/kotlinshiddencosts-benchmarks

consider using the integer constructor instead here, or adding a Iterable constructor.